futures-util/src/future/join_all.rs

```diff
@@ -137,3 +137,28 @@ impl<Item> Future for JoinAll<Item>
         }
     }
 }
+
+impl<F: IntoFuture> FromIterator<F> for JoinAll<F::Future> {
+    fn from_iter<T: IntoIterator<Item=F>>(iter: T) -> Self {
+        join_all(iter)
+    }
+}
+
+#[test]
+fn join_all_from_iter() {
```

Comment:
This test should go to `/tests/all.rs` I think, but those tests are not tested at all in the 0.2 branch.